### DISCUSSION TITLE
If condition isn't matching release branches

### DISCUSSION THREAD

**Discussion by mfarag1** [community/community#27078] [Actions]
<p>Hello.</p>
<p>I have a github action workflow such that this is roughly what it looks like:</p>
<pre><code class="lang-auto">on: 
pull_request:
  branches:
  - dev
  - 'release/**'
  - main
jobs:
  job:
    if: |
      (github.base_ref == 'dev' || github.base_ref == 'release/**' || github.base_ref == 'main')
steps:
  # the rest of the script
</code></pre>
<p>I have tried every permutation of how release/** can be mentioned there, but github keeps skipping this job when I try merging a branch to a release branch.</p>

**Comment by airtower-luna** [comment]
<p>The problem is that <code>==</code> is a simple equality check, not a glob operation, so it won’t match unless your branch is literally named <code>release/**</code>.</p>
<p>Why do you need the <code>if</code>? The <code>pull_request.branches</code> condition should already ensure the workflow (and its jobs) runs only if the PR is targeting one of those branches.</p>

**Comment by mfarag1** [comment, marked as answer]
<p>There are other jobs I’d include in this workflow. I want to limit the job based on branch names. But I discovered a solution to this issue using contains function. Everything would be the same as I have in the example above, but checking for release would look like this:<br>
contains(github.base_ref, ‘release’)</p>
